src/Elasticsearch.Net/Utf8Json/MemoryPool.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+
+namespace Elasticsearch.Net.Utf8Json
+{
+	internal static class MemoryPool
```

Comment:
nitpick: can we rename this to something obnoxious like `ArrayPoolProxy`. Since we have `MemoryStreamFactory` and `ArrayPool` as two distinct ways to lease memory `MemoryPool` sounds like it could belong to either.

Reply:
`ByteArrayPool`?

Reply:
`ArrayPoolProxy` is good, renamed